src/bcf-toolkit/Converter/Bcf30/SchemaConverterToBcf21.cs

```diff
@@ -1,11 +1,296 @@
+    builder
+      .SetIfcProject(from.IfcProject)
+      .SetIfcSpatialStructureElement(from.IfcSpatialStructureElement)
+      .SetIsExternal(from.IsExternal)
+      .SetFileName(from.Filename)
+      .SetDate(from.Date)
+      .SetReference(from.Reference);
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.BimSnippet? ConvertBimSnippet(Model.Bcf30.BimSnippet? from) {
+    if (from is null) {
+      return null;
+    }
+
+    var builder = new BimSnippetBuilder();
+    builder
+      .SetSnippetType(from.SnippetType)
+      .SetIsExternal(from.IsExternal)
+      .SetReference(from.Reference)
+      .SetReferenceSchema(from.ReferenceSchema);
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.TopicDocumentReference ConvertDocumentReference(
+    Model.Bcf30.DocumentReference from) {
+    var builder = new DocumentReferenceBuilder();
+    var isExternal = string.IsNullOrEmpty(from.Url);
+    builder
+      .SetGuid(from.Guid)
+      .SetDescription(from.Description)
+      .SetIsExternal(isExternal)
+      .SetReferencedDocument(isExternal ? from.Url : from.DocumentGuid);
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.Comment ConvertComment(Model.Bcf30.Comment from) {
+    var builder = new CommentBuilder();
+
+    builder
+      .SetGuid(from.Guid)
+      .SetDate(from.Date)
+      .SetAuthor(from.Author)
+      .SetCommentProperty(from.CommentProperty)
+      .SetViewPointGuid(from.Viewpoint.Guid)
+      .SetModifiedDate(from.ModifiedDate)
+      .SetModifiedAuthor(from.ModifiedAuthor);
+
+    return builder.Build();
+  }
+
+  private static Model.Bcf21.ViewPoint ConvertViewPoint(
+    Model.Bcf30.ViewPoint from) {
+
+    var viewPoint = new ViewPoint();
+
+    viewPoint.VisualizationInfo =
+      ConvertVisualizationInfo(from.VisualizationInfo);
+
+    return viewPoint;
+  }
+
+  private static Model.Bcf21.VisualizationInfo? ConvertVisualizationInfo(
+    Model.Bcf30.VisualizationInfo? from) {
+    if (from is null)
+      return null;
+
+    var builder = new VisualizationInfoBuilder();
+
+    builder
+      .SetGuid(from.Guid)
+      .AddSelections(
+        from.Components.Selection.Select(ConvertComponent).ToList())
+      .SetVisibility(ConvertVisibility(from.Components.Visibility))
+      .AddBitmaps(from.Bitmaps.Select(ConvertBitmap).ToList())
+      .AddColorings(from.Components.Coloring.Select(ConvertColoring).ToList())
+      .AddLines(from.Lines.Select(ConvertLine).ToList())
+      .AddClippingPlanes(from.ClippingPlanes.Select(ConvertClippingPlane)
+        .ToList())
+      .SetOrthogonalCamera(ConvertOrthogonalCamera(from.OrthogonalCamera))
+      .SetPerspectiveCamera(ConvertPerspectiveCamera(from.PerspectiveCamera));
+    //TODO: SetViewSetupHints data loss
+
+    return builder.Build();
+
+  }
+
+  private static Model.Bcf21.OrthogonalCamera ConvertOrthogonalCamera(
+    Model.Bcf30.OrthogonalCamera from) {
+    var camera = new OrthogonalCamera();
```

Comment:
Orthogonal camera builder should be used, or validation should be do manually.

Reply:
Done.